controller/namedspaces_test.go

```diff
@@ -120,9 +122,125 @@ func (rest *TestNamedSpaceREST) TestSuccessListSpaces() {
+	_, collabspaces := test.ListNamedspacesOK(t, collabSpaceSvc.Context, collabSpaceSvc, collabSpacectrl, testsupport.TestIdentity.Username, nil, nil, nil)
 	assert.True(t, len(collabspaces.Data) > 0)
 	assert.Equal(t, created.Data.Attributes.Name, collabspaces.Data[0].Attributes.Name)
 	assert.Equal(t, created.Data.Attributes.Description, collabspaces.Data[0].Attributes.Description)
 	assert.Equal(t, created.Data.Links.Self, collabspaces.Data[0].Links.Self)
 }
+
+func (rest *TestNamedSpaceREST) TestSuccessSortListSpaces() {
+	t := rest.T()
+	resource.Require(t, resource.Database)
+
+	spaceSvc, spaceCtrl := rest.SecuredSpaceController()
+	identityRepo := account.NewIdentityRepository(rest.DB)
+	identity := testsupport.TestIdentity
+	identity.ProviderType = account.KeycloakIDP
+	err := identityRepo.Create(spaceSvc.Context, &identity)
+	if err != nil {
+		assert.Fail(t, "Failed to create an identity")
+	}
+
+	// create spaces
+	inputSpaces := []string{
+		"Test 24",
+		"Test 22",
+		"Test 23",
+	}
+	// randominze the space names so that they don't clash
+	random := testsupport.CreateRandomValidTestName("")
+	for i := range inputSpaces {
+		inputSpaces[i] += random
+	}
+
+	for _, sn := range inputSpaces {
+		p := newCreateSpacePayload(&sn, nil)
+		test.CreateSpaceCreated(t, spaceSvc.Context, spaceSvc, spaceCtrl, p)
+	}
+
+	// Now try to list the data in various circumstances
+	tests := []struct {
+		name      string
+		sortParam *string
+		want      []string
+	}{
+		{
+			name: "no sort param given",
+			want: []string{
+				"Test 23",
+				"Test 22",
+				"Test 24",
+			},
+		},
+		{
+			name:      "sort in ascending using name",
+			sortParam: ptr.String("name"),
+			want: []string{
+				"Test 22",
+				"Test 23",
+				"Test 24",
+			},
+		},
+		{
+			name:      "sort in descending using name",
+			sortParam: ptr.String("-name"),
+			want: []string{
+				"Test 24",
+				"Test 23",
+				"Test 22",
+			},
+		},
+		{
+			name:      "sort in ascending using created at",
+			sortParam: ptr.String("created"),
+			want: []string{
+				"Test 24",
+				"Test 22",
+				"Test 23",
+			},
+		},
+		{
+			name:      "sort in descending using created at",
+			sortParam: ptr.String("-created"),
+			want: []string{
+				"Test 23",
+				"Test 22",
+				"Test 24",
+			},
+		},
+		{
+			name:      "sort in ascending using updated at",
+			sortParam: ptr.String("updated"),
+			want: []string{
+				"Test 24",
+				"Test 22",
+				"Test 23",
+			},
+		},
+		{
+			name:      "sort in descending using updated at",
+			sortParam: ptr.String("-updated"),
+			want: []string{
+				"Test 23",
+				"Test 22",
+				"Test 24",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			svc, ctrl := rest.SecuredNamedSpaceController(testsupport.TestIdentity)
+			_, spaces := test.ListNamedspacesOK(t, svc.Context, svc, ctrl,
+				testsupport.TestIdentity.Username, nil, nil, tt.sortParam)
+			assert.True(t, len(spaces.Data) == len(tt.want))
```

Comment:
it looks good when we know the space, in this case we don't know spaces of both.

Also internally it will again call len on one of the objects list. So this sort of becomes readable compared to this other approach.